SoliditySnippetPlugin/src/plugin.js

```diff
@@ -0,0 +1,176 @@
+
+    extension.call('editor','getCurrentFile', [], function (error, result) {
+      extension.call('editor','getFile', [result[0]], function (error1, result1) {
+
+        insertImport(result1[0], result[0], sigURL, false)
+    })
+  })
+ })
+ document.querySelector('input#string').addEventListener('click', function () {
+
+   extension.call('editor','getCurrentFile', [], function (error, result) {
+     extension.call('editor','getFile', [result[0]], function (error1, result1) {
+
+       insertImport(result1[0], result[0], stringURL, false)
+   })
+ })
+})
+
+document.querySelector('input#zepplinImport').addEventListener('click', function () {
+  zepplinURL = openzBaseURL + document.getElementById('path').value
+
+  extension.call('editor','getCurrentFile', [], function (error, result) {
+    extension.call('editor','getFile', [result[0]], function (error1, result1) {
+
+      insertImport(result1[0], result[0], zepplinURL, false)
+  })
+  })
+})
+}
+
+async function insertImport(code, currentPath, toImport, isLocal){
+  code = await code.split('\n')
+  if(isLocal){
+    await code.splice(1,0,"import \"./" + toImport + ".sol\"\;")
+  }else{
+    await code.splice(1,0,"import \"" + toImport + "\"\;")
+  }
+
+  currentCode = await code.join('\n')
+  await extension.call('editor', 'setFile', [currentPath, currentCode], function (error, result) { console.log(error, result) })
+}
+
+async function projectParser(fetchReturn, start, end, code, currentPath){
+  let currentCode = await code
+  let data = await fetchReturn
+  data = await data.text()
+  data = data.split('\n')
+
+  let elementsAtEnd = await data.length - end
+  await data.splice(0,start-2) //minus two since one is at the start and we want to include start line
+  await data.splice(end - start, elementsAtEnd)
+
+  for(i = data.length; i >=0; i = i - 1){
+      await currentCode.splice(document.getElementById('insertionLine').value, 0, data[i])
+  }
+
+  currentCode = await currentCode.join('\n')
+  console.log(currentPath)
+  console.log(currentCode)
+  await extension.call('editor', 'setFile', [currentPath, currentCode], function (error, result) { console.log(error, result) })
+}
+
+async function fileParser(fetchReturn, start, end, contractName, hasFunction) {
+  let data = await fetchReturn
+  data = await data.text()
+  data = data.split('\n')
+
+  let len = await data.length
+  if(end === 0){
+    end = len
+  }
+  let elementsAtEnd = await len - end
+  await data.splice(1,start-2) //minus two since one is at the start and we want to include start line
+  await data.splice(end - start, elementsAtEnd)
+
+  if(start>0){
+    await data.splice(1,0, " contract " + contractName + " {")
+  }
+  if(end<len){
+    await data.splice(data.length-1,0," } ")
+  }
+
+  if(hasFunction){
+    await data.splice(2,0, document.getElementById('functionName').value)
+    await data.splice(data.length-1,0," } ")
+  }
+
+  let path = await "browser/" + contractName +".sol"
```

Comment:
that will only work for the `browser` explorer. 
But you can use `extension.call('editor','getCurrentFile' ... )` to get the current path